libs/core/src/components/sage-button/test/sage-button.spec.tsx

```diff
@@ -58,7 +58,22 @@ describe('sage-button', () => {
     const svg = root.shadowRoot.querySelector('svg');
     expect(svg).not.toBeNull();
   });
-});
-
-
 
+  it('runs `handleClick` method when clicked on from inside a form', async () => {
```

Comment:
Are we testing the `handleClick` or the `reset`?  This feels more like a `reset` test.

Reply:
We're testing the entirety of the `handleClick` method with this test. We're setting a type other than `button` on our `sage-button` and nesting it into a form. This will run all the if logic within `handleClick`:
1. checking button type
2. checking if `sage-button` has a shadow DOM
3. finding the form its nested within
4. generating a fake button to click inside the form for us